scan-command/src/main/java/io/ballerina/scan/internal/ProjectAnalyzer.java

```diff
@@ -69,68 +117,160 @@ private Consumer<DocumentId> analyzeDocument(Module module, ScannerContextImpl s
+        PackageResolution packageResolution = project.currentPackage().getResolution();
+        ResolvedPackageDependency rootPkgNode = new ResolvedPackageDependency(project.currentPackage(),
+                PackageDependencyScope.DEFAULT);
+        List<Package> directDependencies = packageResolution.dependencyGraph().getDirectDependencies(rootPkgNode)
+                .stream().map(ResolvedPackageDependency::packageInstance).toList();
+        Map<String, List<Rule>> externalAnalyzers = new HashMap<>();
+        for (Package pkgDependency : directDependencies) {
+            PackageManifest pkgManifest = pkgDependency.manifest();
+            String org = pkgManifest.org().value();
+            String name = pkgManifest.name().value();
+            String pluginName = org + FORWARD_SLASH + name;
+            if (pkgManifest.compilerPluginDescriptor().isEmpty() || !analyzerDescriptors.contains(pluginName)) {
+                continue;
+            }
 
-        // TODO: Implement the logic to pass scanner contexts to compiler plugins and perform external static analysis.
+            CompilerPluginDescriptor pluginDesc = pkgManifest.compilerPluginDescriptor().get();
+            List<URL> jarUrls = pluginDesc.dependencies().stream().map(dependency -> {
+                        try {
+                            return Path.of(dependency.getPath()).toUri().toURL();
+                        } catch (MalformedURLException ex) {
+                            throw new RuntimeException(ex);
+                        }
+            }).toList();
 
-        List<Issue> externalIssues = new ArrayList<>(scannerContextList.size());
-        for (ScannerContext scannerContext : scannerContextList) {
-            ReporterImpl reporter = (ReporterImpl) scannerContext.getReporter();
-            externalIssues.addAll(reporter.getIssues());
+            URLClassLoader ucl = URLClassLoader.newInstance(jarUrls.toArray(URL[]::new),
+                    this.getClass().getClassLoader());
+            InputStream resourceAsStream = ucl.getResourceAsStream(RULES_FILE);
+            if (resourceAsStream == null) {
+                continue;
+            }
+
+            String output;
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(resourceAsStream,
+                    StandardCharsets.UTF_8))) {
+                output = reader.lines().collect(Collectors.joining());
+            } catch (IOException ex) {
+                throw new RuntimeException(ex);
+            }
+
+            List<Rule> externalRules = new ArrayList<>();
+            Gson gson = new Gson();
+            JsonElement element = gson.fromJson(output, JsonElement.class);
+            if (!element.isJsonArray()) {
+                throw new ScanToolException(DiagnosticLog.error(DiagnosticCode.INVALID_JSON_FORMAT, RULES_FILE,
+                        pluginName, gson.toJson(element)));
+            }
+
+            JsonArray ruleArray = element.getAsJsonArray();
```

Comment:
I would break this up to methods where possible. 

For example `getRuleJsonArray`. This makes the code more readable giving an overview of what happens and we'll have to check the actual logic only if required.